base/exports.jl

```diff
@@ -484,13 +484,15 @@ export
     zeta,
 
 # arrays
+    allocate_for,
```

Comment:
:( Maybe this can be combined with `similar`? If not, at least needs a better name.